### PR TITLE
[Fix: issue 266] Added script Component

### LINKED ISSUE #266: Allow Script Tags
Sometimes the user may need to provide JS scripts (for example, adding a Google Analytics script to their site). We need a clean way to allow this.

### PR DESCRIPTION
**Pull request by wassafshahzad**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Description
Adds a new component which can be used to add script tags.

### Related Tickets
Closes #266

### CODE REVIEW

**Comment by ElijahAhianyo** [comment]
Nice. We could probably use some unit tests for this, esp the render method

**Review by ElijahAhianyo — COMMENTED** [review]
Nice. We could probably use some unit tests for this, esp the render method

**Comment by wassafshahzad** [comment]
> Nice. We could probably use some unit tests for this, esp the render method

Sure, Is there an example testcase I could look into ?

**Comment by ElijahAhianyo** [comment]
> > Nice. We could probably use some unit tests for this, esp the render method
> 
> Sure, Is there an example testcase I could look into ?

Yes, there are a couple tests in tests/components. A few of them are:
- https://github.com/pynecone-io/pynecone/blob/main/tests/components/datadisplay/test_datatable.py
- https://github.com/pynecone-io/pynecone/blob/main/tests/components/layout/test_cond.py
- https://github.com/pynecone-io/pynecone/blob/main/tests/components/test_component.py

**Comment by wassafshahzad** [comment]
> > > Nice. We could probably use some unit tests for this, esp the render method
> > 
> > 
> > Sure, Is there an example testcase I could look into ?
> 
> Yes, there are a couple tests in tests/components. A few of them are:
> 
> * https://github.com/pynecone-io/pynecone/blob/main/tests/components/datadisplay/test_datatable.py
> * https://github.com/pynecone-io/pynecone/blob/main/tests/components/layout/test_cond.py
> * https://github.com/pynecone-io/pynecone/blob/main/tests/components/test_component.py

Will look into it tomorrow

File: pynecone/components/base/link.py [hunk 1]
```diff
@@ -19,3 +21,33 @@ def _render(self) -> Tag:
             href=self.href,
             rel=self.rel,
         )
+
+class ScriptTag(Component):
+    """A component that creates a script tag with the speacified type and source."""
+
+    type: Var[str]
```
Comment (picklelo) [inline pynecone/components/base/link.py:28]: Can we add comments for each of the props explaining them

File: pynecone/components/base/link.py [hunk 2]
```diff
@@ -19,3 +21,33 @@ def _render(self) -> Tag:
             href=self.href,
             rel=self.rel,
         )
+
+class ScriptTag(Component):
```
Comment (picklelo) [inline pynecone/components/base/link.py:25]: We can add the first prop as
```python
tag = "script"
```

File: pynecone/components/base/link.py [hunk 3]
```diff
@@ -19,3 +21,33 @@ def _render(self) -> Tag:
...
+    """A component that creates a script tag with the speacified type and source."""
+
+    type: Var[str]
+
+    source: Var[str]
+
+    integrity: Optional[Var[str]] = None
```
Comment (picklelo) [inline pynecone/components/base/link.py:32]: We don't need default values, they will automatically be set to `None`

File: pynecone/components/base/link.py [hunk 4]
```diff
@@ -19,3 +21,33 @@ def _render(self) -> Tag:
...
+
+    referrer_policy: Optional[Var[str]] = None
+
+    is_async: Optional[Var[bool]] = None
+    
+    defer: Optional[Var[bool]] = None
+
+
+
+    def _render(self) -> Tag:
```
Comment (picklelo) [inline pynecone/components/base/link.py:44]: Once you add the `tag = "script"` prop for the component, we can get rid of this function I believe. The `super()._render` method should handle this.

**Review by picklelo — CHANGES_REQUESTED** [review]
Great work! Just a couple minor comments - also make sure to run the linters to make sure the build passes: https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md

File: pynecone/components/base/link.py [hunk 5]
```diff
@@ -19,3 +21,56 @@ def _render(self) -> Tag:
...
             rel=self.rel,
         )
+
+
+class ScriptTag(Component):
+    """A component that creates a script tag with the speacified type and source.
+
+
+    Args:
+        type: This attribute indicates the type of script represented.
```
Comment (picklelo) [inline pynecone/components/base/link.py:31]: In a follow up, let's move these comments to be directly above each prop, like in the other component examples. This lets us automatically generate documentation for it on our main website.

**Review by picklelo — APPROVED** [review]
Looks good!

File: pynecone/components/base/link.py (hunk 3 above)
Comment (fnwinter) [reply pynecone/components/base/link.py]: Is the source correct? Could it be src ?

**Pull request merged**